.github/workflows/publish.yml

```
@@ -10,24 +10,150 @@ permissions:
   id-token: write
 
 jobs:
-  build-and-publish:
+  guard:
+    name: Guard (tag points to main?)
     runs-on: ubuntu-latest
+    outputs:
+      publish: ${{ steps.check.outputs.publish }}
     steps:
-      - name: Check out repository
+      - name: Checkout (shallow)
         uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Check if tag commit is on main
+        id: check
+        env:
+          DEFAULT_BRANCH: ${{ github.event.repository.default_branch }}
+        run: |
+          git fetch origin $DEFAULT_BRANCH --depth=1
+          if git merge-base --is-ancestor "$GITHUB_SHA" "origin/$DEFAULT_BRANCH"; then
+            echo "publish=true" >> "$GITHUB_OUTPUT"
+            echo "Tag commit is on $DEFAULT_BRANCH; will publish."
+          else
+            echo "publish=false" >> "$GITHUB_OUTPUT"
+            echo "Tag commit is NOT on $DEFAULT_BRANCH; skipping publish."
+          fi
+
+  build-wheels:
+    name: Build native wheels
+    needs: guard
+    if: needs.guard.outputs.publish == 'true'
+    continue-on-error: ${{ matrix.allow-failure == true }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-14, windows-latest]
+        python-version: ['3.12', '3.13']
+        include:
+          - os: ubuntu-latest
+            python-version: '3.14'
+            allow-failure: true
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up Rust
+        uses: dtolnay/rust-toolchain@stable
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install build backend
```

[P2] Pin manylinux to manylinux2014 to match compatibility intent

The Linux wheel build uses `manylinux: auto`, which can produce `manylinux_2_28` wheels on newer toolchains. This reduces compatibility compared to `manylinux2014` (as implied by the PR title) and may prevent installation on older glibc systems. Pin `manylinux: manylinux2014` to ensure broader compatibility. Suggestion:
```suggestion
          manylinux: manylinux2014
```